compliance/aws/instances_without_fnm_agent/aws_instances_not_running_flexnet_inventory_agent.pt

```diff
@@ -0,0 +1,226 @@
+      field "computerName", xpath(col_item, "ComputerName")
+      field "instanceCloudID", xpath(col_item, "R2_fa9c6ab728d0ffd494dcba715bddad62_ComputerToCloudServiceInstance_InstanceCloudID")
+    end
+  end
+end
+
+datasource "ds_formatted_instances", type: "javascript" do
+  run_script $js_format_instances,  $ds_aws_ec2_instances_list, $ds_fnms_report, $param_exclude_tags
+end
+
+
+###############################################################################
+# Scripts
+###############################################################################
+
+#Formatter script for the Instance data
+script "js_format_instances", type: "javascript" do
+  parameters "ds_aws_ec2_instances_list","ds_fnms_report", "param_exclude_tags"
+  result "result"
+  code <<-EOS
+    var result = [];
+	
+    fnms_instance_cloud_id=[];
+    _.each(ds_fnms_report, function(instance){
+      fnms_instance_cloud_id.push(instance["instanceCloudID"]);
+    })
+	
+    var param_exclude_tags_lower=[];
+    for(var i=0; i < param_exclude_tags.length; i++){
+      param_exclude_tags_lower[i]=param_exclude_tags[i].toString().toLowerCase();
+    }
+
+    _.each(ds_aws_ec2_instances_list, function(instance){
+      var tags = instance['tags'];
+      var isTagMatched=false
+      var tagKeyValue=""
+      for(var k=0; k < tags.length; k++){
+        tag = tags[k]
+        //Check, if the tag present in entered param_exclude_tags, ignore the EC2 instance if the tag matches/present.
+        if((param_exclude_tags_lower.indexOf((tag['tagKey']).toLowerCase()) !== -1) || (param_exclude_tags_lower.indexOf((tag['tagKey']+'='+tag['tagValue']).toLowerCase()) !== -1)){
+          isTagMatched = true;
+        }
+        if((tag['tagValue']).length > 0){
+          tagKeyValue = tagKeyValue + ', '+ tag['tagKey']+'='+tag['tagValue']
+        }else{
+          tagKeyValue = tagKeyValue + ', '+ tag['tagKey']
+        }
+      }
+      if(!isTagMatched && fnms_instance_cloud_id.indexOf(instance["instance_id"]) === -1){
+        result.push({
+          instance_id: instance['instance_id'],
+          instance_type: instance['instance_type'],
+          region: instance['zone'][0]['availability_zone'],
+          instance_state: instance['instance_state'],
+          tags: (tagKeyValue.slice(2))
+        })
+      }
+    })
+  EOS
+end
+
+###############################################################################
+# Escalations
+###############################################################################
+
+escalation "report_instances" do
+  email $param_email
+end
+
+###############################################################################
+# Policy
+###############################################################################
+
+policy "policy_long_running_instances" do
+  validate $ds_formatted_instances do
+    summary_template "{{ rs_project_name }} (Account ID: {{ rs_project_id }}): {{ len data }} EC2 instances are missing in FlexNet Manager."
```

Comment:
Grammar check.  the word **are** should probably be removed

Reply:
Removed